src/kinetics/Kinetics.cpp

```diff
@@ -143,8 +143,13 @@ std::pair<size_t, size_t> Kinetics::checkDuplicates(bool throw_err) const
                     continue; // No overlap in third body efficiencies
                 }
             } else if (R.type() == "three-body") {
-                ThirdBody& tb1 = dynamic_cast<ThreeBodyReaction&>(R).third_body;
-                ThirdBody& tb2 = dynamic_cast<ThreeBodyReaction&>(other).third_body;
+                auto& R1 = dynamic_cast<ThreeBodyReaction&>(R);
+                auto& R2 = dynamic_cast<ThreeBodyReaction&>(other);
+                if (R1.specified != R2.specified) {
+                    continue; // No overlap as one third body is explicitly specified
+                }
+                ThirdBody& tb1 = R1.third_body;
+                ThirdBody& tb2 = R2.third_body;
```

Comment:
Does the duplicate reaction handling actually need to change? I guess I would be inclined to regard the two reactions below as unmarked duplicates, and I think that without this change, that would already be the case.
```yaml
- equation: H + O2 + H2O <=> HO2 + H2O
  rate-constant: {A: 1.126e+19, b: -0.76, Ea: 0.0}
- equation: H + O2 + M <=> HO2 + M
  rate-constant: {A: 1.126e+19, b: -0.76, Ea: 0.0}
  type: three-body
  default-efficiency: 0
  efficiencies: {H2O: 1}
```

Reply:
This was also my main comment in looking at this PR. I agree that those reactions should be treated as unmarked duplicates; likewise something like `H + O2 + O2` and `H + 2 O2` should be treated as a duplicate (although I'd be surprised if that particular case wasn't already treated as a duplicate).

Reply:
I think that’s a valid point. Regarding how this is treated internally, third body `O2`‘S are already detected, so how this is written doesn’t matter. My initial thought had mainly been that a user may *want* to keep those as unmarked duplicates, but it’s certainly safer the other way.

Reply:
Done. Took me a while, as I didn't realize that a corresponding unit test would need to write a file (!) ... see #1019